plugins/rbac/src/utils/conditional-access-utils.ts

```diff
@@ -0,0 +1,129 @@
+};
+
+export const nestedConditionButtons = conditionButtons.filter(
+  button => button.val !== 'condition',
+);
+
+export const extractNestedConditions = (
+  conditions: Condition[],
+  criteriaTypes: string[],
+  nestedConditions: Condition[],
+) => {
+  conditions.forEach(c => {
+    criteriaTypes.forEach(ct => {
+      if (Object.keys(c).includes(ct)) {
+        nestedConditions.push(c);
+      }
+    });
+  });
+};
+
+export const getDefaultRule = (selPluginResourceType: string) => ({
+  rule: '',
+  resourceType: selPluginResourceType,
+  params: {},
+});
+
+export const useConditionsFormRowStyles = makeStyles(theme => ({
+  conditionRow: {
+    padding: '20px',
+    border: `1px solid ${theme.palette.border}`,
+    borderRadius: '4px',
+    backgroundColor: theme.palette.background.default,
+    '& input': {
+      color: `${theme.palette.textContrast}!important`,
+      '&:-internal-autofill-selected, &:-webkit-autofill, &:-webkit-autofill:hover, &:-webkit-autofill:focus, &:-webkit-autofill:active':
+        {
+          WebkitBoxShadow: `0 0 0px 1000px ${theme.palette.background.paper} inset`,
+          WebkitTextFillColor: `${theme.palette.textContrast}!important`,
+          caretColor: `${theme.palette.textContrast}!important`,
+        },
+    },
+    '& button': {
+      textTransform: 'none',
+    },
+  },
+  nestedConditionRow: {
+    padding: '20px',
+    marginLeft: theme.spacing(3),
+    border: `1px solid ${theme.palette.border}`,
+    borderRadius: '4px',
+    backgroundColor: theme.palette.background.default,
+    '& input': {
+      backgroundColor: `${theme.palette.background.paper}!important`,
+    },
+  },
+  criteriaButtonGroup: {
+    backgroundColor: theme.palette.background.paper,
+    width: '80%',
+  },
+  criteriaButton: {
+    width: '100%',
+    padding: `${theme.spacing(1)}px !important`,
+  },
+  nestedConditioncriteriaButtonGroup: {
+    backgroundColor: theme.palette.background.paper,
+    width: '60%',
+    height: '100%',
+  },
+  addRuleButton: {
+    display: 'flex !important',
+    color: theme.palette.primary.light,
+    textTransform: 'none',
+  },
+  addNestedConditionButton: {
+    display: 'flex !important',
+    color: theme.palette.primary.light,
+    textTransform: 'none',
+  },
+  removeRuleButton: {
+    color: theme.palette.grey[500],
+    flexGrow: 0,
+    alignSelf: 'baseline',
+    marginTop: theme.spacing(3.3),
+  },
+  removeNestedRuleButton: {
+    color: theme.palette.grey[500],
+    flexGrow: 0,
+    alignSelf: 'baseline',
+  },
+  radioGroup: {
+    margin: theme.spacing(1),
+  },
+  radioLabel: {
+    marginTop: theme.spacing(1),
+  },
+}));
+
+export const calculateConditionIndex = (
+  conditionRow: ConditionsData,
+  nestedConditionIndex: number,
+  criteria: string,
+): number => {
+  const simpleRulesCount =
+    criteria === criterias.not
+      ? 0
+      : (conditionRow[criteria as keyof Condition] as Condition[]).filter(
+          (e: Condition) => 'rule' in e,
```

Comment:
@ciiay, maybe: 
```suggestion
          (e: Condition) => e && 'rule' in e,
```

because, I faced with bug here: 

https://github.com/user-attachments/assets/aff4268c-79fd-4a4c-ac11-736cdacf143a

Reply:
Possible workaround is "?? []": : (conditionRow[criteria as keyof Condition] as Condition[] ?? []).filter(